lib/compiler-singlepass/src/compiler.rs

```diff
@@ -256,6 +257,11 @@ impl Compiler for SinglepassCompiler {
             debug: dwarf,
         })
     }
+
+    fn get_cpu_features_used(&self, cpu_features: &EnumSet<CpuFeature>) -> EnumSet<CpuFeature> {
+        let used = CpuFeature::AVX | CpuFeature::SSE42 | CpuFeature::LZCNT | CpuFeature::BMI1;
```

Comment:
I didn't want to go on tracking the kind of instructions used for now. It would make the emitter more complicated, so I prefer to opt for a "potentially used" approach for now.

Reply:
Yeah, that makes sense. We can be conservative now and always refine things later on if it becomes an issue.